### DISCUSSION TITLE
Slightly worse bounding box detections with EfficientDet D0 when compared with TF OD API.

### DISCUSSION THREAD

**Discussion by sovit-123** [rwightman/efficientdet-pytorch#146] [Q&A] [answered]
I trained both this repo's EfficientDet D0 and TF Object Detection API D0. After the same number of training steps, the TF version is giving better FPS and also somewhat better detections. Can anyone provide some thought on this? Any particular reason. What I could be doing wrong. I used the `tf_efficientdet_d0` from this repo.

I used the same optimizer with momentum, the same steps for warm restart. Still, unable to figure out the reason.

**Comment by rwightman** [comment, marked as answer]
@sovit-123 I matched training results against the original official releases (mine was 33.6 theirs 33.5 for D0) of the Tensorflow weights with earlier training runs here. At some point they updated their weights (34.2 D0) with better trained versions, and updated code with updated Keras variants of the model and I never quite figured out what changed on their end in the training recipe to allow that improvement. They didn't appear to mention anything specific, the models remained structurally the same. A lot of extras were added to their training code like bbox IOU etc that didn't appear to be enabled.

If anyone finds the magic that allowed for their jump I'd be happy to update hparams or training code.

**Reply by sovit-123** [reply]
Thanks a lot for replying. Yes, there are a lot of hyperparameters to be considered and matched while training. In fact, I am currently using this repo for custom detection and trying my best to match the params. If I find something that really seems right, then I will surely notify. By the way thanks for the amazing repo. It is very clear that this repo can be extended with proper support for a lot more years to come.
